Extend .NET SDK docs with info about DaprWorkflowClient (#4957)
dapr/docs: sdkdocs/dotnet/content/en/dotnet-sdk-docs/dotnet-workflow/dotnet-workflowclient-usage.md

{{% /alert %}} 

### Singleton Registration

> By default, the `AddDaprWorkflow` method will register the `DaprWorkflowClient` and associated services using a singleton lifetime. This means

Reviewer: ```suggestion
By default, the `AddDaprWorkflow` method registers the `DaprWorkflowClient` and associated services using a singleton lifetime. This means
```

> that the services will be instantiated only a single time.

Reviewer: ```suggestion
that the services are instantiated only a single time.
```

await app.RunAsync();
```

### Create a `DaprWorkflowClient` instance

> To create a `DaprWorkflowClient` instance in your `Program.cs`, retrieve it from the `ServiceProvider`:

Reviewer 2: This guidance is entirely accurate for console applications which rarely use anything but Program.cs, but the vast majority of users are going to be using ASP.NET Core - I'd suggest a simple demonstration showing dependency injection into a controller via the primary constructor or something to reflect that more common use case as well.

```csharp

> await using var scope = host.Services.CreateAsyncScope();

Reviewer 2: I'd have to double check, but I don't believe this is awaitable.

Author: I copied it from the .NET workflow examples: https://github.com/dapr/dotnet-sdk/blob/master/examples/Workflow/WorkflowTaskChaining/Program.cs

Reviewer 2: Oops - per [this link](https://learn.microsoft.com/en-us/dotnet/api/microsoft.extensions.dependencyinjection.serviceproviderserviceextensions.createasyncscope?view=net-10.0-pp) it is not awaitable. I'll create a PR to fix the example too.

Reviewer 2: I recommend putting a `[FromServices]` in front of `DaprWorkflowClient` as that's commonly used in the minimal API setting these days to clearly articulate where any given argument is coming from.

Author: Great, both changes are done
